### PR TITLE
WalletConnect add network select

### PR DESCRIPTION
**Pull request by AmazingBeerBelly**
CleanShot 2020-10-09 at 17 39 34@2x

### CODE REVIEW

File: src/pages/wallet/wallet.connect/connecting.js [hunk 1]
```diff
@@ -129,6 +139,20 @@ class WalletConnecting extends Component {
           <Text style={styles.content}>{strings('page.wallet.walletconnect.agreeTwo')}</Text>
         </View>
 
+        <View style={styles.block}>
+          <Loc style={styles.title} text="page.wallet.walletconnect.advancedOptions" />
+          <SwitchRow
+            value={isTestnet}
+            text={strings('page.wallet.addCustomToken.testnet')}
+            titleStyle={styles.switchTitleStyle}
```
Comment (imstar15) [inline src/pages/wallet/wallet.connect/connecting.js:147]: titleStyle，这个属性我并没有定义啊
如果你需要的话，可以往里加，叫textStyle吧

**Review by imstar15 — CHANGES_REQUESTED** [review]

File: src/pages/wallet/wallet.connect/connecting.js (hunk 1 above)
Comment (AmazingBeerBelly) [reply src/pages/wallet/wallet.connect/connecting.js]: 这是多余的代码，已经删掉了。

File: src/common/translations/pt-BR.json [hunk 2]
```diff
@@ -123,7 +123,8 @@
       "walletconnect": {
         "title": "Wallet Connect",
         "emptyWalletError": "Carteira vazia",
-        "selectAvailableWallet": "Por favor selecione uma carteira que tenha pelo menos um ativo na rede principal da rsk",
+        "selectMainnetAsset": "Selecione uma carteira que tenha pelo menos um ativo na rede principal da rsk",
+        "selectTestnetAsset": "Selecione uma carteira que tenha pelo menos um ativo Testnet rsk",
```
Comment (bitcoinuser) [inline src/common/translations/pt-BR.json:127]:
```suggestion
        "selectTestnetAsset": "Selecione uma carteira que tenha pelo menos um ativo na rede de testes da rsk",
```

File: src/common/translations/pt-BR.json [hunk 3]
```diff
@@ -169,7 +171,11 @@
         "type": "Tipo",
         "from": "De",
         "to": "Para",
-        "data": "Dados"
+        "data": "Dados",
+        "networkQuestion": {
+          "title": "Rede",
+          "body": "A testenet é uma cadeia alternativa de blocos RSK, a ser usada para testar."
```
Comment (bitcoinuser) [inline src/common/translations/pt-BR.json:177]:
```suggestion
          "body": "A rede de testes ou testnet é uma blockchain alternativa da RSK usada exclusivamente para realização de testes."
```

File: src/common/translations/pt-BR.json [hunk 4]
```diff
@@ -213,7 +219,8 @@
       "addCustomToken": {
         "title": "Adicionar ativos personalizados",
         "address": "Endereço do contrato do ativo",
-        "mainnet": "Rede principal"
+        "mainnet": "Mainnet",
```
Comment (bitcoinuser) [inline src/common/translations/pt-BR.json:222]:
```suggestion
        "mainnet": "Rede prinicipal",
```

File: src/common/translations/pt-BR.json [hunk 5]
```diff
@@ -213,7 +219,8 @@
...
+        "testnet": "Testnet"
```
Comment (bitcoinuser) [inline src/common/translations/pt-BR.json:223]:
```suggestion
        "testnet": "Rede de testes"
```

**Review by bitcoinuser — CHANGES_REQUESTED** [review]
Changes Portuguese Brazilian

**Review by chrisli30 — CHANGES_REQUESTED** [review]

**Review by chrisli30 — APPROVED** [review]

**Review by chrisli30 — APPROVED** [review]

**Comment by chrisli30** [comment]
Need to test against Sovryn Testnet 
https://test.sovryn.app/

**Review by imstar15 — APPROVED** [review]

File: src/pages/wallet/wallet.connect/modal/transaction.js [hunk 6]
```diff
@@ -38,7 +38,7 @@ export default function TransactionModal({
   const {
     value, from, to, data, gasLimit, gasPrice,
   } = txData;
-  const amount = Rsk3.utils.hexToNumber(value);
+  const amount = value;
```
Comment (chrisli30) [inline src/pages/wallet/wallet.connect/modal/transaction.js:41]: Add comment here why amount is value(and not Rsk3.utils.hexToNumber(value))

**Review by chrisli30 — APPROVED** [review]
Looking good. Only one comment left.

File: src/pages/wallet/wallet.connect/index.js [hunk 7]
```diff
@@ -224,16 +226,40 @@ class WalletConnectPage extends Component {
...
+    const { isTestnet } = this.state;
     const { navigation: { state: { params: { wallet } } } } = this.props;
     const { coins } = wallet;
-    const ethChainCoins = _.filter(coins, (coin) => coin.symbol !== 'BTC' && coin.type === 'Mainnet');
-    if (!_.isEmpty(ethChainCoins)) {
-      return {
-        address: Rsk3.utils.toChecksumAddress(ethChainCoins[0].address),
-        privateKey: ethChainCoins[0].privateKey,
-      };
+    const net = isTestnet ? 'Testnet' : 'Mainnet';
```
Comment (chrisli30) [inline src/pages/wallet/wallet.connect/index.js:232]: net? should be called "network" ...

File: src/pages/wallet/wallet.connect/index.js [hunk 8]
```diff
@@ -30,7 +30,9 @@ import screenHelper from '../../../common/screenHelper';
 import color from '../../../assets/styles/color';
 import appActions from '../../../redux/app/actions';
 
-const { MAINNET } = NETWORK;
+const { MAINNET, TESTNET } = NETWORK;
+const WALLET_CONNECTING = 'WalletConnecting';
+const WALLET_CONNECTED = 'WalletConnected';
```
[the remark this answers is not in the document]
Comment (chrisli30) [reply src/pages/wallet/wallet.connect/index.js:36]: Address this issue?

**Review by chrisli30 — APPROVED** [review]
There are two comments left in this PR. Please address those.

**Review by chrisli30 — APPROVED** [review]

File: src/pages/wallet/wallet.connect/index.js [hunk 9]
```diff
@@ -688,10 +699,44 @@ class WalletConnectPage extends Component {
...
+      return (
+        <WalletConnecting
+          approve={this.approveSession}
+          reject={this.rejectSession}
+          address={address}
+          dappName={name}
+          dappUrl={url}
+          isTestnet={isTestnet}
+          onSwitchValueChanged={async () => {
+            await this.setState({ isTestnet: !isTestnet });
```
Comment (imstar15) [inline src/pages/wallet/wallet.connect/index.js:718]: 这个回调函数提出来
await  this.setState()，这样的写法不确定能不能行。
一般我们用this.setState({}, () => {})

**Review by imstar15 — CHANGES_REQUESTED** [review]

File: src/pages/wallet/wallet.connect/index.js (hunk 9 above)
Comment (AmazingBeerBelly) [reply src/pages/wallet/wallet.connect/index.js:718]: await this.setState()可以的，跟this.setState({}, () => {})是一个作用

**Review by chrisli30 — APPROVED** [review]

**Review by imstar15 — APPROVED** [review]

**Pull request merged**